src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    if (!this.hidden) {
+      this.hidden = true;
+    }
+  };
+};
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (!beast.hidden && !(beast instanceof Carnivore)) {
+      beast.health -= 50;
+    }
+
+    if (beast.health === 0) {
+      Animal.alive = Animal.alive.filter(creature => creature !== beast);
```

Comment:
And change condition
Let's move the `delete` method into a instance of the `Animal` class

```
if (animal.health <= 0) {
  Animal.removeFromAlive(animal);
}
```